components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -142,7 +142,7 @@ protected boolean requiredIDToken(Map<String, String> authenticatorProperties) {
      */
 
     protected String getAuthenticateUser(AuthenticationContext context, Map<String, Object> jsonObject,OAuthClientResponse token) {
-        return (String) jsonObject.get("sub");
+        return (String) jsonObject.get("upn");
```

Comment:
This does not seem to be the correct fix. With this change all other OpenID connect scenarios which rely on the subject being picked from 'sub' attribute will fail.